src/components/tl/mlx5/mcast/tl_mlx5_mcast_helper.c

```diff
@@ -571,23 +553,28 @@ ucc_status_t ucc_tl_mlx5_mcast_modify_rc_qps(ucc_tl_mlx5_mcast_coll_context_t *c
     return UCC_OK;
 }
 
-ucc_status_t ucc_tl_mlx5_fini_mcast_group(ucc_tl_mlx5_mcast_coll_context_t *ctx,
-                                          ucc_tl_mlx5_mcast_coll_comm_t    *comm)
+ucc_status_t ucc_tl_mlx5_leave_mcast_group(ucc_tl_mlx5_mcast_coll_context_t *ctx,
+                                           ucc_tl_mlx5_mcast_coll_comm_t    *comm)
 {
     char        buf[40];
     const char *dst;
+    int         i;
 
-    dst = inet_ntop(AF_INET6, &comm->mcast.groups[0].mcast_addr, buf, 40);
-    if (NULL == dst) {
-        tl_error(comm->lib, "inet_ntop failed");
-        return UCC_ERR_NO_RESOURCE;
-    }
+    for (i = 0; i < comm->mcast_group_count; i++) {
+        if (comm->mcast.groups[i].mcast_addr.sin6_flowinfo != 0) {
+            dst = inet_ntop(AF_INET6, &comm->mcast.groups[i].mcast_addr, buf, 40);
+            if (NULL == dst) {
+                tl_error(comm->lib, "inet_ntop failed");
+                return UCC_ERR_NO_RESOURCE;
+            }
 
-    tl_debug(ctx->lib, "mcast leave: ctx %p, comm %p, dgid: %s", ctx, comm, buf);
+            tl_debug(ctx->lib, "mcast leave: ctx %p, comm %p, dgid: %s group %d", ctx, comm, buf, i);
 
-    if (rdma_leave_multicast(ctx->id, (struct sockaddr*)&comm->mcast.groups[0].mcast_addr)) {
-        tl_error(comm->lib, "mcast rmda_leave_multicast failed");
-        return UCC_ERR_NO_RESOURCE;
+            if (rdma_leave_multicast(ctx->id, (struct sockaddr*)&comm->mcast.groups[i].mcast_addr)) {
```

Comment:
Should try to leave all groups even if some fail, and return the most severe error